tools/c7n_mailer/c7n_mailer/utils.py

```diff
@@ -365,6 +369,18 @@ def kms_decrypt(config, logger, session, encrypted_field):
         return None
 
 
+def decrypt(config, logger, session, encrypted_field):
+    if config.get(encrypted_field):
+        if is_azure_cloud(config):
+            from c7n_mailer.azure.utils import azure_decrypt
+            return azure_decrypt(config, logger, session, encrypted_field)
```

Comment:
The transport get instantiated when we determine which processor we use. Therefore, I would prefer if we removed the cloud provider logic and essentially just set the "decrypter" or "secrets manager" as part of instantiating the transports. This would really remove the need for the utility methods with switch logic for the cloud providers.

Reply:
You can just set the default behavior if you do not set a decrypter is to return the config value, which is completely cloud agnostic.

Reply:
given this is actually a noop, let's remove it till its actually doing something, dead abstractions are dead weight.

Reply:
I was going to add KeyVault support in a foreseeable future. I'd prefer to keep it as is now, this code runs once per execution, so it doesn't add any significant overhead.

Reply:
I commented out the code here & in tests

Reply:
generally its better to just leave them out / remove them, and just implement the feature when you implement the feature. littering a code base with unused/vestigal parts even commented out has a cognitive/maintenance cost to others. ie unused abstractions are dead weight.